### PR TITLE
DM-14136: Enable numpydoc support for ip_isr

### PR DESCRIPTION
**Pull request by DavidStaker**

### CODE REVIEW

File: python/lsst/ip/isr/isrTask.py [hunk 1]
```diff
@@ -318,23 +320,7 @@ class IsrTaskConfig(pexConfig.Config):
...
-     - @ref ip_isr_isr_Initialize
-     - @ref ip_isr_isr_IO
-     - @ref ip_isr_isr_Config
-     - @ref ip_isr_isr_Debug
-
-
-    @section ip_isr_isr_Purpose Description
-
-    The process for correcting imaging data is very similar from camera to camera.
+    """The process for correcting imaging data is very similar from camera to camera.
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: I would have the first line here be Apply common instrument signature correction algorithms to a raw frame.

File: python/lsst/ip/isr/assembleCcdTask.py [hunk 2]
```diff
@@ -158,26 +141,40 @@ def __init__(self, **kwargs):
...
-          <DD> The input exposures to be assembled do not adhere to the required format.
-        </DL>
-
-        @throws RuntimeError with the following string:
-
-        <DL>
-          <DT> No ccd detector found
-          <DD> The detector set on the input exposure is not set.
-        </DL>
+        """Assemble a set of amps into a single CCD size image
```
Comment (ebellm) [inline python/lsst/ip/isr/assembleCcdTask.py]: period at the end.

File: python/lsst/ip/isr/isrTask.py [hunk 3]
```diff
@@ -345,78 +331,79 @@ class IsrTask(pipeBase.CmdLineTask):
...
-                                  corrected with the retrieved calibration data;
-                                  should not be modified in this method.
-        @return a pipeBase.Struct with fields containing kwargs expected by run()
-         - bias: exposure of bias frame
-         - dark: exposure of dark frame
-         - flat: exposure of flat field
-         - defects: list of detects
-         - fringeStruct: a pipeBase.Struct with field fringes containing
-                         exposure of fringe frame or list of fringe exposure
+        """Retrieve necessary frames for instrument signature removal
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: add period

File: python/lsst/ip/isr/isrTask.py [hunk 4]
```diff
@@ -478,32 +465,53 @@ def run(self, ccdExposure, bias=None, linearizer=None, dark=None, flat=None, def
             opticsTransmission=None, filterTransmission=None,
             sensorTransmission=None, atmosphereTransmission=None,
             crosstalkSources=None):
-        """!Perform instrument signature removal on an exposure
+        """Perform instrument signature removal on an exposure
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: period

File: python/lsst/ip/isr/isrTask.py [hunk 5]
```diff
@@ -682,19 +690,28 @@ def convertIntToFloat(self, exposure):
         return newexposure
 
     def biasCorrection(self, exposure, biasExposure):
-        """!Apply bias correction in place
+        """Apply bias correction in place
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: period

File: python/lsst/ip/isr/isrTask.py [hunk 6]
```diff
@@ -682,19 +690,28 @@ def convertIntToFloat(self, exposure):
...
+        exposure :
+            exposure to process
+        biasExposure :
+            bias exposure of same size as exposure
         """
         isrFunctions.biasCorrection(exposure.getMaskedImage(), biasExposure.getMaskedImage())
 
     def darkCorrection(self, exposure, darkExposure, invert=False):
-        """!Apply dark correction in place
+        """Apply dark correction in place
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: period

File: python/lsst/ip/isr/isrTask.py [hunk 7]
```diff
@@ -760,11 +780,16 @@ def updateVariance(self, ampExposure, amp, overscanImage=None):
             )
 
     def flatCorrection(self, exposure, flatExposure, invert=False):
-        """!Apply flat correction in place
+        """Apply flat correction in place
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: period

File: python/lsst/ip/isr/isrTask.py [hunk 8]
```diff
@@ -775,13 +800,22 @@ def flatCorrection(self, exposure, flatExposure, invert=False):
         )
 
     def getIsrExposure(self, dataRef, datasetType, immediate=True):
-        """!Retrieve a calibration dataset for removing instrument signature
+        """Retrieve a calibration dataset for removing instrument signature
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: period

File: python/lsst/ip/isr/isrTask.py [hunk 9]
```diff
@@ -800,10 +834,14 @@ def getIsrExposure(self, dataRef, datasetType, immediate=True):
         return exp
 
     def saturationDetection(self, exposure, amp):
-        """!Detect saturated pixels and mask them using mask plane config.saturatedMaskName, in place
+        """Detect saturated pixels and mask them using mask plane config.saturatedMaskName, in place
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: period

File: python/lsst/ip/isr/isrTask.py [hunk 10]
```diff
@@ -816,13 +854,20 @@ def saturationDetection(self, exposure, amp):
             )
 
     def saturationInterpolation(self, ccdExposure):
-        """!Interpolate over saturated pixels, in place
+        """Interpolate over saturated pixels, in place
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: period

File: python/lsst/ip/isr/isrTask.py [hunk 11]
```diff
@@ -832,16 +877,20 @@ def saturationInterpolation(self, ccdExposure):
         )
 
     def suspectDetection(self, exposure, amp):
-        """!Detect suspect pixels and mask them using mask plane config.suspectMaskName, in place
+        """Detect suspect pixels and mask them using mask plane config.suspectMaskName, in place
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: period

File: python/lsst/ip/isr/isrTask.py [hunk 12]
```diff
@@ -857,12 +906,18 @@ def suspectDetection(self, exposure, amp):
         )
 
     def maskAndInterpDefect(self, ccdExposure, defectBaseList):
-        """!Mask defects using mask plane "BAD" and interpolate over them, in place
-
-        @param[in,out]  ccdExposure     exposure to process
-        @param[in] defectBaseList a list of defects to mask and interpolate
+        """Mask defects using mask plane "BAD" and interpolate over them, in place
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: period

File: python/lsst/ip/isr/isrTask.py [hunk 13]
```diff
@@ -878,15 +933,18 @@ def maskAndInterpDefect(self, ccdExposure, defectBaseList):
         )
 
     def maskAndInterpNan(self, exposure):
-        """!Mask NaNs using mask plane "UNMASKEDNAN" and interpolate over them, in place
+        """Mask NaNs using mask plane "UNMASKEDNAN" and interpolate over them, in place
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: period

File: python/lsst/ip/isr/isrTask.py [hunk 14]
```diff
@@ -970,23 +1028,30 @@ def overscanCorrection(self, exposure, amp):
         return results
 
     def addDistortionModel(self, exposure, camera):
-        """!Update the WCS in exposure with a distortion model based on camera geometry
+        """Update the WCS in exposure with a distortion model based on camera geometry
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: period

File: python/lsst/ip/isr/isrTask.py [hunk 15]
```diff
@@ -1003,10 +1068,14 @@ def addDistortionModel(self, exposure, camera):
         exposure.setWcs(distortedWcs)
 
     def setValidPolygonIntersect(self, ccdExposure, fpPolygon):
-        """!Set the valid polygon as the intersection of fpPolygon and the ccd corners
+        """Set the valid polygon as the intersection of fpPolygon and the ccd corners
```
Comment (ebellm) [inline python/lsst/ip/isr/isrTask.py]: period

**Comment by timj** [comment]
If this branch has no possibility of being merged can we close the PR?

**Pull request closed without merging**